### PR TITLE
Serialization for ThreadPool objects

### PR DESCRIPTION
**Pull request by peddie**
This commit adds serialization support for `ThreadPool`s.

### CODE REVIEW

File: tests/test_serialize.cc [hunk 1]
```diff
@@ -449,4 +449,22 @@ TEST(test_serialize, test_joint_streamable) {
   oss << dist;
 }
 
+TEST(test_serialize, test_ThreadPool) {
```
Comment (akleeman) [inline tests/test_serialize.cc:452]: nit: might be worth a `nullptr` test as well

**Review by akleeman — APPROVED** [review]
lgtm!

File: tests/test_serialize.cc (hunk 1 above)
Comment (peddie) [reply tests/test_serialize.cc:452]: Will do!

Comment (peddie) [reply tests/test_serialize.cc]: I expanded test coverage substantially, have a look.

**Pull request merged**